### ISSUE TITLE
scalaJS support

### ISSUE THREAD

**Issue by antonkulaga** [nscala-time/nscala-time#82]
At the moment ScalaJS allows to use either javascript Date class or java.util.Date classes. That is why for cross scala-scalajs libraries one has to use only terrible default java.util.Date class :-(

**Comment by kmizu** [comment]
I think so, too.  However, since nscala-time heavily depends on joda-time, unless joda-time for JavaScript (:- is provided, We cannot support scalaJS.

**Comment by wookietreiber** [comment]
Is there a label for **maybe in the future** or **depends on external stuff**? I would attach this to this issue.

**Comment by kmizu** [comment]
Hmmm.  Certainly, there should be such label.

**Comment by kmizu** [comment]
@wookietreiber A little while ago, I created a label `maybe_in_the_future` and I attached it to this issue and I reopened this issue.  Are you satisfied with this?

**Comment by wookietreiber** [comment]
Yeah, sure, looks good.

**Comment by wookietreiber** [comment]
I don't know much about how scala-js works, but just as a thought: Will scala-js support come easier if #85 gets implemented? I.e. is it possible with scala-js to make use of the std java stuff?

**Comment by xuwei-k** [comment]
> Will scala-js support come easier if #85 gets implemented? I.e. is it possible with scala-js to make use of the std java stuff?

I don't think so. scala-js does not support Java8 datetime API yet. see https://github.com/scala-js/scala-js/tree/v0.6.0/javalib/src/main/scala/java

**Comment by mdedetrich** [comment]
Just letting you guys know, that I am creating a port of jodatime to Scala/Scala.js. Currently in very alpha stage (got it to compile!), but that may be another avenue to get nscala-time to work on Scala.js

**Comment by bchazalet** [comment]
@mdedetrich do you have anything to share at this stage? I'm curious

**Comment by mdedetrich** [comment]
@bchazalet As per what was said on gitter, the project is currently compiling, but some parts need to be rewritten (some business logic that used break/continue/continue to label is broken).

When this is done, I will upload it onto github, but more modifications will need to be done to actually get it to work with `Scala.js`.

Thankfully its at a stage where a lot of the grunt work has been done. Expect a push in probably a week

**Comment by mdedetrich** [comment]
Hey guys (and @bchazalet ), just letting you know that i have done an early release of Soda-Time, which is the port of `Joda-Time` to Scala/Scala.js. Its still in alpha stage, but since so many people were asking for it, I released it early

https://github.com/mdedetrich/soda-time

**Comment by bchazalet** [comment]
@mdedetrich  that's great news, I will definitely have a look !

**Comment by xuwei-k** [comment]
> I don't think so. scala-js does not support Java8 datetime API yet. see https://github.com/scala-js/scala-js/tree/v0.6.4/javalib/src/main/scala/java

scala-js will support Java8 time https://github.com/scala-js/scala-js/tree/6a6b4f72b/javalib/src/main/scala/java/time

**Comment by xuwei-k** [comment]
https://github.com/scala-js/scala-js-java-time

**Comment by xuwei-k** [comment]
https://github.com/soc/scala-java-time

**Comment by xuwei-k** [comment]
https://github.com/opt-tech/chronoscala/pull/5

**Comment by nafg** [comment]
@kmizu haven't you seen [js-joda](https://github.com/js-joda/js-joda)?
